app/javascript/components/Greeting.js

```diff
@@ -0,0 +1,55 @@
+import React from "react"
+import PropTypes from "prop-types"
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+
+const GET_GREETINGS_REQUEST = 'GET_GREETINGS_REQUEST';
+const GET_GREETINGS_SUCCESS = 'GET_GREETINGS_SUCCESS';
+
+const getGreetings = () => {
+ return dispatch => {
+   dispatch({ type: GET_GREETINGS_REQUEST });
+   return fetch(`v1/greetings.json`)
+   .then(response => response.json())
+   .then(json => dispatch(getGreetingsSuccess(json)))
+   .catch(error => console.log(error));
```

Comment:
## OPTIONAL 
* Kindly remove all `console.log ` statements from your codebase as it is not good practice you might be exposing sensitive information to users